relayer/relayer.go

```diff
@@ -284,6 +285,62 @@ func (r *Relayer) ReconnectToSubscriber() error {
 	return nil
 }
 
+// Fetch the appropriate message relayer
+// Checks for the following registered keys. Exactly one of these keys should be registered, or none of them.
+// 1. An exact match on sourceBlockchainID, destinationBlockchainID, originSenderAddress, and destinationAddress
+// 2. A match on sourceBlockchainID and destinationBlockchainID, with any originSenderAddress and any destinationAddress
+// 3. A match on sourceBlockchainID and destinationBlockchainID, with a specific originSenderAddress and any destinationAddress
+// 4. A match on sourceBlockchainID and destinationBlockchainID, with any originSenderAddress and a specific destinationAddress
+func (r *Relayer) getMessageRelayer(
+	sourceBlockchainID ids.ID,
+	destinationBlockchainID ids.ID,
+	originSenderAddress common.Address,
+	destinationAddress common.Address,
+) (*messageRelayer, bool) {
+	// Check for an exact match
+	messageRelayerID := database.CalculateRelayerID(
+		sourceBlockchainID,
+		destinationBlockchainID,
+		originSenderAddress,
+		destinationAddress,
+	)
+	if messageRelayer, ok := r.messageRelayers[messageRelayerID]; ok {
+		return messageRelayer, ok
+	}
+
+	// Check for a match on sourceBlockchainID and destinationBlockchainID, with any originSenderAddress and any destinationAddress
+	messageRelayerID = database.CalculateRelayerID(
+		sourceBlockchainID,
+		destinationBlockchainID,
+		common.Address{},
```

Comment:
Could we define a `allAllowedDestinationAddress` const and use that here instead?

Reply:
Added